pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -3178,3 +3179,135 @@ func TestGetWorkloadNameForPod(t *testing.T) {
 		t.Errorf("Expected different workload name\n want: %s\n got: %s", wantWlName, wlName)
 	}
 }
+
+func TestReconciler_PodIsObservedWhenUpdateOrDeleteOpsFail(t *testing.T) {
+
+	reqcount := 0
+	now := time.Now()
+	connRefusedEerrMock := fmt.Errorf("connection refused: %w", syscall.ECONNREFUSED)
+	ctx, _ := utiltesting.ContextWithLog(t)
+
+	basePodWrapper := testingpod.MakePod("pod", "ns").
+		UID("test-uid").
+		Queue("user-queue").
+		Request(corev1.ResourceCPU, "1").
+		Image("", nil)
+
+	pods := []corev1.Pod{
+		*basePodWrapper.
+			Clone().
+			Label("kueue.x-k8s.io/managed", "true").
+			KueueFinalizer().
+			Group("test-group").
+			GroupTotalCount("2").
+			CreationTimestamp(now).
+			Obj(),
+		*basePodWrapper.
+			Clone().
+			Name("pod2").
+			Label("kueue.x-k8s.io/managed", "true").
+			KueueFinalizer().
+			Group("test-group").
+			CreationTimestamp(now.Add(time.Minute)).
+			GroupTotalCount("2").
+			Obj(),
+		*basePodWrapper.
+			Clone().
+			Name("excessPod").
+			Label("kueue.x-k8s.io/managed", "true").
+			KueueFinalizer().
+			Group("test-group").
+			CreationTimestamp(now.Add(time.Minute * 2)).
+			GroupTotalCount("2").
+			Obj(),
+	}
+
+	wl := *utiltesting.MakeWorkload("test-group", "ns").
+		PodSets(
+			*utiltesting.MakePodSet("60bc72d3", 2).
+				Request(corev1.ResourceCPU, "1").
+				Obj(),
+		).
+		Queue("user-queue").
+		OwnerReference(corev1.SchemeGroupVersion.WithKind("Pod"), "pod", "test-uid").
+		OwnerReference(corev1.SchemeGroupVersion.WithKind("Pod"), "pod2", "test-uid").
+		OwnerReference(corev1.SchemeGroupVersion.WithKind("Pod"), "excessPod", "test-uid").
+		ReserveQuota(utiltesting.MakeAdmission("cq").AssignmentPodCount(1).Obj()).
+		Admitted(true).
+		Obj()
+
+	clientBuilder := utiltesting.NewClientBuilder()
+	if err := SetupIndexes(ctx, utiltesting.AsIndexer(clientBuilder)); err != nil {
+		t.Fatalf("Could not setup indexes: %v", err)
+	}
+
+	for i := range pods {
+		clientBuilder = clientBuilder.WithObjects(&pods[i])
+	}
+
+	kcBuilder := clientBuilder.
+		WithStatusSubresource(&wl).
+		WithInterceptorFuncs(interceptor.Funcs{
+			Update: func(ctx context.Context, client client.WithWatch, obj client.Object, opts ...client.UpdateOption) error {
+				if reqcount == 0 {
+					reqcount += 1
+					return connRefusedEerrMock
+				}
+				reqcount = 0
+				return client.Update(ctx, obj, opts...)
+			},
+			Delete: func(ctx context.Context, client client.WithWatch, obj client.Object, opts ...client.DeleteOption) error {
+				if reqcount == 0 {
+					reqcount += 1
+					return connRefusedEerrMock
+				}
+				reqcount = 0
+				return client.Delete(ctx, obj, opts...)
+			},
+		})
+
+	kClient := kcBuilder.Build()
+	if err := kClient.Create(ctx, &wl); err != nil {
+		t.Fatalf("Could not create workload: %v", err)
+	}
+
+	recorder := record.NewBroadcaster().NewRecorder(kClient.Scheme(), corev1.EventSource{Component: "test"})
+	reconciler := NewReconciler(kClient, recorder)
+
+	reconcilerImpl := reconciler.(*Reconciler)
```

Comment:
This casting could be eliminated if we don't assert expecatations directly, as suggested in the comment lower down.

Reply:
Done 👍🏻